### ISSUE TITLE
Error during mod installation

### ISSUE THREAD

**Issue by TrefKorr** [fifty-six/Scarab#118]
### Scarab Version

1.19.0

### Hollow Knight Version

1.5.80.11835

### Steps to Reproduce

<img width="338" alt="image" src="https://user-images.githubusercontent.com/108542994/209476722-b915d2a2-e960-432a-8124-61b9d1ace49f.png">
Click an "Install" button on any mod.

### Expected Behaviour

Succesful installation of a mod.

### Actual Behaviour

<img width="1248" alt="image" src="https://user-images.githubusercontent.com/108542994/209476744-83152604-4808-4426-8ad9-40e36c12a9bf.png">

### Logs

None.

**Comment by fifty-six** [comment]
I need logs for this, which you should in fact have

**Comment by TrefKorr** [comment]
> I need logs for this, which you should in fact have

Ok, where can I get logs?

**Comment by Athofry** [comment]
I have the same problem, I think it is related to the fact that scarab does not find the Hollow Knight folders correctly

**Comment by TrefKorr** [comment]
> I have the same problem, I think it is related to the fact that scarab does not find the Hollow Knight folders correctly

No, this problem occurs only in new versions of HK; in the older ones everything is fine, so I think it is a compatibility problem rather then problem with finding files.

**Comment by fifty-six** [comment]
The logs are in `%appdata%\HKModInstaller`. This shouldn't be version related as HK hasn't even had an update in a while and I always test with current patch.

**Comment by Athofry** [comment]
[5cc4847a-c7e7-4018-8a86-0e58b4f12001ModInstaller.log](https://github.com/fifty-six/Scarab/files/10327893/5cc4847a-c7e7-4018-8a86-0e58b4f12001ModInstaller.log)
These are the logs you say?

**Comment by fifty-six** [comment]
Yeah, thanks. Could I get the other files in that folder? Could you also show me what your Managed folder looks like if that's not too much trouble? If you go to open mods it's the folder containing mods (so you just go one back)

**Comment by fifty-six** [comment]
Yeah, that's the right file. If you have the installer open, there's an `Open Mods Folder` button that you can and in the top navbar from there it'll say something like `... > Managed > Mods`, it'd be useful to have a picture of the Managed folder. Just to double check, though it sounds like you tried this, verifying your game integrity didn't change anything?

**Comment by Athofry** [comment]
![Scarab 20_01_2023 21_27_50](https://user-images.githubusercontent.com/121686857/213799481-ecd7a30b-5356-4adc-ade3-947f7a1d31c1.png)
![Hollow_Knight_Data 20_01_2023 21_28_41](https://user-images.githubusercontent.com/121686857/213799665-e55ebdd5-bf72-49fd-bd10-4b1912d5b1a5.png)

-¿Are these the photos you need?

-And yes, I have checked the integrity of the game from steam. In fact, as I already mentioned, if I don't do it, the game doesn't load and the screen stays black, and I have to download that file again.

-I don't know if the problem has to do with an incompatibility, but before I bought the game I played it pirated, and I didn't have the same problem, I didn't get to play with the mods because my version was old and I had a problem with unity for that reason, but I didn't have the current problem about the file path.

-And if you need something else, tell me.

**Comment by Cavernosa** [comment]
>-¿Are these the photos you need?

I think they meant you to show what's inside the Managed folder, where there's a bunch of `.dll`s, not the folder itself

Btw try renaming Hollow_Knight_Data to hollow_knight_Data to see if it does something. I was pretty sure the steam default is the latter

**Comment by fifty-six** [comment]
Yeah, inside the Managed folder is what I was looking for - checking to see if there's an `Assembly-CSharp.dll` and/or `Assembly-CSharp.dll.m`

**Comment by Athofry** [comment]
![Managed 27_01_2023 17_06_39](https://user-images.githubusercontent.com/121686857/215133623-70eaa1f9-2bd3-4334-9f9f-b5448345e995.png)
![Managed 27_01_2023 17_06_46](https://user-images.githubusercontent.com/121686857/215133628-712c3614-16df-414e-afaa-fabaabdfb7a2.png)
![Managed 27_01_2023 17_06_25](https://user-images.githubusercontent.com/121686857/215133634-230b3f61-8b16-4c80-bc4d-152204f234ab.png)

-Oh, I'm sorry. Okay, you were referring to this, right?

**Comment by fifty-six** [comment]
Could you disable the API, verify your game integrity, and then relaunch Scarab? I think that might fix your issue, though if it does I do want to add safeguards for this.

**Comment by Athofry** [comment]
-I can't disable the API because I don't have any first mod

**Comment by shadowibm** [comment]
Did u find any solution?

PD: Also, I've found that, when u press install on any mod it deletes the "Assembly-CSharp.dll" file, in case it helps
